content/posts/0.21-release.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "Pyodide 0.21.0 release"
+date: 2022-08-02
+tags: ["release", "announcement"]
+author: "Hood Chatham on behalf of the Pyodide team"
+showToc: true
+TocOpen: false
+hidemeta: false
+comments: false
+# description: ""
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+---
+
+Pyodide 0.21 is a major release focused on improvements to the systems for
+building and loading packages. We added support for out of tree package builds
+which should soon make it much easier for people to build binary packages that
+we do not include in our distribution. We added support for Rust packages and
+made major progress in supporting CMake packages. We also added 34 new packages
+including a large number of popular packages, for example bitarray,
+opencv-python, shapely, and xgboost.
+
+We moved from Emscripten v2.0.27 to Emscripten v3.1.14 which includes almost all
+of the patches we need (we still apply two patches). There were also many minor
+improvements and bugfixes to the Pyodide module and type conversions.
+
+
+
+For the complete list of changes see the
+[changelog](https://pyodide.org/en/stable/project/changelog.html#version-0-21-0).
+
+## Out of tree binary package builds for Pyodide
+
+The most exciting new feature in Pyodide 0.21.0 is that it adds experimental
+support for building binary packages for Pyodide "out of tree".
+
+Pyodide has so far used a monolothic collection of package recipes inspired by
+Conda to build binary extensions. Using a collection of package recipes has
+significant advantages, allowing us to maintain patches and test all of the
+packages together. However, it also has several downsides:
+
+1. It requires all packages to be maintained in a monorepo.
+
+2. It doesn't allow packages to support Pyodide in their CI.
+
+3. There is no convenient way for someone to use a personal-use binary extension
+   with Pyodide.
+
+4. There is no good way to have several different versions of the same package.
+   For instance, this would be very helpful for using Pyodide to make
+   interactive documentation for old versions of a package.
+
+We can build most packages out of tree, though this feature is still a bit
+experimental. We will continue using the collection of package recipes for our
+in-tree builds. Our goal is to add Pyodide to the CI of key packages, for
```

Comment:
Maybe it would be good to add an example here to make it more concrete. And also emphasize that this no longer requires downloading the pyodide repo.

For instance, here is a minimal setup to build a package wheel in Github Actions CI,
```yaml
runs-on: ubuntu-latest
    steps:
      -  uses: actions/checkout@v3
      - uses: actions/setup-python@v4
        with:
          python-version: 3.10.2
      - uses: mymindstorm/setup-emsdk@v11
        with:
          version: 3.1.14
      - run: pip install pyodide-build==0.21.0
      - run: pyodide build
```

Opened https://github.com/pyodide/pyodide/issues/2949